autorosters/autorosters_main.py

```diff
@@ -0,0 +1,270 @@
+            for team in current_teams:
+                for player in self.rosters_data[team]["players"].keys():
+                    for role in self.rosters_data[team]["players"][player]["games_by_role"]:
+                        self.rosters_data[team]["players"][player]["games_by_role"][role] += ","
+        for team_data in self.rosters_data.values():
+            for player in team_data["players"].values():
+                for role, role_data in player["games_by_role"].items():
+                    player["games_by_role"][role] = role_data[:-1]
+
+    @staticmethod
+    def get_order(rosters_data):
+        role_numbers = {
+            "Top": 1,
+            "Jungle": 2,
+            "Mid": 3,
+            "Bot": 4,
+            "Support": 5
+        }
+        sorted_teams = sorted(list(rosters_data.keys()))
+        sorted_data = {"teams": sorted_teams, "players": {}}
+        for team, team_data in rosters_data.items():
+            team_players = {}
+            for player, player_data in team_data["players"].items():
+                team_players[player] = role_numbers[player_data["roles"][0]]
+            sorted_data["players"][team] = sorted(team_players.items(), key=lambda x: x[1])
+        return sorted_data
+
+    @staticmethod
+    def concat_args(data):
+        ret = ''
+        lookup = data
+        if type(data) == dict:
+            lookup = []
+            for k, v in data.items():
+                lookup.append({k: v})
+
+        for pair in lookup:
+            pair: dict
+            for key in pair.keys():
+                if pair[key] is None:
+                    ret = ret + '|{}='.format(key)
+                else:
+                    ret = ret + '|{}={}'.format(key, str(pair[key]))
+        return ret
+
+    def make_output(self, players_data):
+        output = self.PAGE_HEADER.format(self.tabs)
+        sorted_data = self.get_order(self.rosters_data)
+        for team in sorted_data["teams"]:
+            players_text = ""
+            for player in sorted_data["players"][team]:
+                player = player[0]
+                if players_data.get(player):
+                    player_data = self.concat_args(players_data[player])
+                else:
+                    player_data = self.concat_args([{"player": player}])
+                player_roles_data = self.concat_args(self.rosters_data[team]["players"][player]["roles_data"])
+                player_games_by_role = self.concat_args(self.rosters_data[team]["players"][player]["games_by_role"])
+                players_text += self.PLAYER_TEXT.format(player_data, player_roles_data, player_games_by_role)
+            teamsvs = self.concat_args(self.rosters_data[team]["teamsvs"])
+            output += self.TEAM_TEXT.format(team, teamsvs, players_text)
+        return output
+
+    def save_page(self, output):
+        page = self.site.client.pages[f"{self.overview_page}/Team Rosters"]
```

Comment:
Since we're not doing coaches here, I don't want to overwrite the actual live data with this edit. Let's instead save to a user page (do it for whoever owns the bot account) so `User:{user name of the auth credentials}/Team Rosters Sandbox`

Then after this is done running, instead of "Okay done!" let's print:

Okay, done! **Remember the generated content has no coaches!**
Here is the sandbox page with the new content: {}
Here is where you should copy it: {}